### PR TITLE
[CBRD-22206] JSON_TABLE function

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/CBRD-22206

JSON table draft implementation. Grammar, name resolving, query optimization, XASL generation and scan manager are affected.

Evaluating scan predicate is still in progress.

Code also needs some cleanup.

### CODE REVIEW

**Comment by ribram** [comment]
Currently, filtering is done after a complete row is scanned. It should be enough for first iteration.

Optimizing filters to JSON table node level will be done in a separate task.

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/query/scan_manager.c (1 incorrect indentation(s) found)
```
  4362 	{
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

File: src/parser/csql_grammar.y [hunk 1]
```diff
@@ -24069,7 +24114,134 @@ vacuum_stmt
...
+    | ERROR_
+      {{
+        $$.m_behavior = JSON_TABLE_THROW_ERROR;
+        $$.m_default_value = NULL;
+      DBG_PRINT}}
+    | DEFAULT CHAR_STRING
+      {{
+        $$.m_behavior = JSON_TABLE_DEFAULT_VALUE;
+        $$.m_default_value = parser_alloc (this_parser, sizeof (struct db_value));
+        db_make_string ($$.m_default_value, $2);
```
Comment (eseokoh) [inline src/parser/csql_grammar.y]: I'm not sure to use `db_make_string` is safe. 
It might be better to generate a parse node and convert it to db_value.

**Comment by eseokoh** [comment]
please find two examples of http://jira.cubrid.org/browse/CBRD-22206
server crashes with the first and client does with the second.

**Comment by ribram** [comment]
I have pushed some serious changes on the xasl_to_stream/stream_to_xasl modules.

XASL generation and packing/unpacking had some bugs hard to detect. This kind of bugs usually are easily generated and hard to find.

In order to make investigation easier, I added a debug check, that builds structure from packed data right after executing process. This should detect usual bugs early.

Unfortunately, making stx_build functionality visible to xasl_to_stream required many changes. I moved necessary functions and structures to xasl_stream.hpp/xasl_stream.cpp.

stream_to_xasl cannot be included in CS_MODE, since it depends on SERVER_MODE/SA_MODE restricted structures.

**Pull request merged**